proposals/collection-literals.md

```diff
@@ -207,57 +203,37 @@ Through the use of the [`init`](#init-methods) modifier, existing APIs can direc
 
     Here, if `b` is false, it is not required that any value actually be constructed for the empty literal since it would immediately be spread into zero values in the final literal.
 
-
 ## Natural type
 [natural-type]: #natural-type
 
 In the absence of a *constructible collection target type*, a non-empty literal can have a *natural type*.
 
 The *natural type* is determined using the [*best common type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#116315-finding-the-best-common-type-of-a-set-of-expressions) algorithm.
 
-A [*natural element type*](#natural-element-type) `T` is first determined.  If that cannot be determined, the literal has no *natural type*.  If `T` can be determined and it is some `KeyValuePair<TKey, TValue>`, then the *natural type* of the collection is `Dictionary<TKey, TValue>`; otherwise, the *natural type* of the collection is `List<T>`.
+A *natural element type* `T` is first determined from the *best common type* of:
+* The *type* of each element `e_n`, and
+* The *iteration type* of each element `..s_n`.
```

Comment:
i just realized (and this may be ok), that this means the following won't be legal:

```c#
var  v = ["foo", .. x ? [null] : []];
```

I would be fine seeing if this is relevant in practice and if we need to potentially refine how spread interacts with this algorithm.  Specifically if it has no iteration type, we might say that it contributes nothing.  Then the T is figured out, then the T helps target type the spread.  This may be getting too complex and in the weeds though about a non-relevant scenario in practice.